Data check not running -> Bug introduced by #649 or anything not installed right on noreaster?

@jteresco @yakra 

[michih@noreaster ~]$ /fast/tm/datacheck
Tue Oct  8 15:33:44 EDT 2024
/fast/tm/datacheck: Running as common datacheck, setting appropriate defaults
/fast/tm/datacheck: updating TM repositories
remote: Enumerating objects: 531, done.
remote: Counting objects: 100% (531/531), done.
remote: Compressing objects: 100% (187/187), done.
remote: Total 531 (delta 350), reused 520 (delta 343), pack-reused 0 (from 0)
Receiving objects: 100% (531/531), 698.40 KiB | 11.26 MiB/s, done.
Resolving deltas: 100% (350/350), completed with 22 local objects.
From https://github.com/michihdeu/HighwayData
   f398f825a3..b801b88ebb  master     -> origin/master
Updating f398f825a3..b801b88ebb
Fast-forward
 datacheckfps.csv                                 |  5 +++++


...

Already up to date.
/fast/tm/datacheck: creating directories
/fast/tm/datacheck: gathering repo head info
HighwayData @ b801b88ebb076a4f2815dff3e94a27272c35871b
UserData @ 235404ce0e1378834fd6188763029b42bc2b091a
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
DataProcessing @
/fast/tm/datacheck: creating .time files
gmake: *** No rule to make target 'list/10ACDawg.list.time'.  Stop.

--- a/siteupdate/siteupdate.sh
+++ b/siteupdate/siteupdate.sh
@@ -318,9 +318,15 @@ fi
 echo "$0: gathering repo head info"
 echo $repo '@' `(cd $tmbasedir/$repo; git show -s --pretty=%H)` | tee $indir/$logdir/siteupdate.log
 echo UserData '@' `(cd $tmbasedir/UserData; git show -s --pretty=%H)` | tee -a $indir/$logdir/siteupdate.log
-if [[ "$0" == "/fast/tm/datacheck" ]]; then cd /fast/tm/DataProcessing; fi
-echo DataProcessing '@' `git show -s --pretty=%H` | tee -a $indir/$logdir/siteupdate.log
-if [[ "$0" == "/fast/tm/datacheck" ]]; then cd - > /dev/null; fi
+if [[ "$0" == "/fast/tm/datacheck" ]]; then
+    cd /fast/tm/DataProcessing/.git
+    dphead=`sed 's~ref: ~~' HEAD`
+    if [ -e "$dphead" ]; then dphead=`cat "$dphead"`; fi
+    cd - > /dev/null
+else
+    dphead=`git show -s --pretty=%H`
+fi
+echo DataProcessing '@' "$dphead" | tee -a $indir/$logdir/siteupdate.log
 
 echo "$0: creating .time files"
 mkdir -p $tmbasedir/UserData/time_files/$listext

